### PR TITLE
Use Container.Config.Shell instead of hardcoded…

### PR DESCRIPTION
**Pull request by vdemeester**
… for healthcheck. It make the code a little cleaner and more future/usage proof. *Hopefully it's all green :angel:*

/cc @justincormack @talex5 @thaJeztah @cpuguy83 

:frog: 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: daemon/health.go [hunk 1]
```diff
@@ -325,3 +321,13 @@ func min(x, y int) int {
 	}
 	return y
 }
+
+func getShell(config containertypes.Config) []string {
+	if len(config.Shell) != 0 {
+		return config.Shell
+	}
+	if runtime.GOOS != "windows" {
```
Comment (vdemeester) [inline daemon/health.go]: This is duplicated with `builder/dockerfile/builder_unix.go` (and its windows counter-part), we might want to extract that 👼

File: daemon/health.go [hunk 2]
```diff
@@ -325,3 +321,13 @@ func min(x, y int) int {
 	}
 	return y
 }
+
+func getShell(config containertypes.Config) []string {
```
Comment (vdemeester) [inline daemon/health.go]: Hum I like the idea of `Shell()` method on `Config` :angel:
Right now `Shell` is only used while building to set the right `Entrypoint` — it's not used during *runtime*. If `Shell` is empty, it means it has not be set and thus we should use the default behavior (i.e. either `Container.Entypoint` or the current values).

**Comment by crosbymichael** [comment]
LGTM

**Comment by vieux** [comment]
LGTM

**Pull request merged**